src/Features/Core/Portable/Completion/CompletionService.ProviderManager.cs

```diff
@@ -45,7 +46,23 @@ public ProviderManager(CompletionService service, IAsynchronousOperationListener
                         CancellationToken.None);
             }
 
-            public IReadOnlyList<Lazy<CompletionProvider, CompletionProviderMetadata>> GetLazyImportedProviders()
+            private ImmutableDictionary<string, CompletionProvider> LoadImportedProvidersAndCreateNameMap()
+            {
+                var builder = ImmutableDictionary.CreateBuilder<string, CompletionProvider>();
+
+                foreach (var lazyImportedProvider in GetLazyImportedProviders())
+                    builder[lazyImportedProvider.Value.Name] = lazyImportedProvider.Value;
+
+#pragma warning disable CS0618
+                // We need to keep supporting built-in providers for a while longer since this is a public API.
```

Comment:
do we want to deprecate this API?  If so we should probably track and link here

Reply:
Yup. But it's still used by TS. I can add the link in a follow up.